### ISSUE TITLE
Incorrect sequence of calling callbacks `on_take` & `on_put` on swapping items in inventory

### ISSUE THREAD

**Issue by alek13** [minetest/minetest#13486]
##### Minetest version
```
5.7.0
```

##### OS / Hardware
Operating system: Kubuntu 22.04.2 LTS

##### Summary
When swapping items in inventory, MT calls:
 - at first `on_put`
 - and then `on_take`
 
 It will be better if MT calls it in reverse order:
  - `on_take`
  - `on_put`

##### Steps to reproduce
For example, with LOTT game:
1. Install [LOTT](https://content.minetest.net/packages/Amaz/lordofthetest/)
2. * _also you can add debug prints into log [here](https://github.com/minetest-LOTR/Lord-of-the-Test/blob/32eab389274593f91f7f1625d6846bc5f43143e7/mods/lottarmor/armor.lua#L424)_
3. Open inventory, creative tab, search&take "Mithril boots" and "Training boots"
4. Open Main tab and put on Mithril one
5. Take ~wooden~ Training one and swap it with Mithril

As result:
 - you have no boots on preview in inventory
 - you have no boots on player model (F7)
 - and also you have no protection
![image](https://user-images.githubusercontent.com/1972329/236680131-883d089e-8f9e-4761-8423-7dd72c101297.png)

**Comment by SmallJoker** [comment]
PR already exists but waits on another PR which waits for its author #12595
